cms-gem-daq-project/cmsgemos
# Online DB configuration objects

## Description


This PR is to track my progress on DB configuration objects (#213).

Currently implemented:
* Python script to generate `xsd` templates and class definitions from a list of fields.
* Integration of the script into the `Makefile` for the VFAT3 chip configuration.
* XMLSerializationData class representing the structure of XML documents in C++
* Function to turn XMLSerializationData into a DOM tree
* Function that fills an XMLSerializationData object with the contents of a DOM tree
* Tests of serialization functionality:
     * Create a DOM tree and check that it isn't empty
     * Create a DOM tree and validate it against the schema
     * Create a DOM tree, load it and check that we get back the initial object

### Types of changes


## Motivation and Context


#213, #214 and #215 for VFAT3 chip and channel configuration data.

## How Has This Been Tested?



* Checked that a very basic code importing the generated class compiles
* The `xml` file in the repo can be checked using the generated `xsd` template
* Tests compile and yield the expected results
* The tests have been run under Valgrind and no leaks are found 

## Checklist:


- [x] I have updated the documentation accordingly. [Link to documentation](http://homepage.iihe.ac.be/~lmoureau/GEM/cmsgemos/doc/build/html/md_gemonlinedb_gemonlinedb.html)
